warehouse/macaroons/caveats.py

```diff
@@ -83,6 +103,7 @@ def __init__(self, macaroon, context, principals, permission):
 
     def verify(self, key):
         self.verifier.satisfy_general(V1Caveat(self))
+        self.verifier.satisfy_general(ExpiryCaveat(self))
```

Comment:
I was expecting to see a test change as a result of this, but it doesn't look like we're testing the `Verifier` class that well. Could you add a test that captures the change in behavior that this line introduces?

Reply:
This was a great call -- adding these tests ([3e60c4b](https://github.com/pypa/warehouse/pull/11122/commits/3e60c4b51e856a966a4a79259d57cf4ad27eed53)) caught a bug that would have caused us to unconditionally fail all expiry caveat verifications. I've refactored the code to avoid exception-driven failures in caveats going forwards, since that's the root of the problem.